sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -66,11 +105,29 @@ public HttpMessage CreateMessage()
         /// <param name="message">The <see cref="HttpMessage"/> to send.</param>
         /// <param name="cancellationToken">The <see cref="CancellationToken"/> to use.</param>
         /// <returns>The <see cref="ValueTask"/> representing the asynchronous operation.</returns>
-        public ValueTask SendAsync(HttpMessage message, CancellationToken cancellationToken)
+        public async ValueTask SendAsync(HttpMessage message, CancellationToken cancellationToken)
         {
             message.CancellationToken = cancellationToken;
             AddHttpMessageProperties(message);
-            return _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1));
+
+            if (message.Policies == null || message.Policies.Count == 0)
+            {
+                await _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1)).ConfigureAwait(false);
```

Comment:
Let's split this method in two. One that's not async and has the hot path `_pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1)).ConfigureAwait(false);` and the second one with the `else` branch and that can be async.

We'll save a state machine allocation on the hot path this way.